src/string.rs

```diff
@@ -1,8 +1,30 @@
-use core::{cmp::Ordering, fmt, fmt::Write, hash, iter, ops, str};
+//! A fixed capacity [`String`](https://doc.rust-lang.org/std/string/struct.String.html).
+
+use core::{
+    char::{self, DecodeUtf16Error},
+    cmp::Ordering,
+    fmt,
+    fmt::Write,
+    hash, iter, ops, str,
+};
 
 use crate::Vec;
 
-/// A fixed capacity [`String`](https://doc.rust-lang.org/std/string/struct.String.html)
+/// The error type for [`String`] operations.
+#[derive(Debug, PartialEq, Eq)]
+pub enum Error {
```

Comment:
this should be called `FromUtf16Error` and its documentation should indicate that's it's only used in the `from_utf16` method. otherwise, it sounds like this `Error` type is going to be used in all of String's fallible API, which is not the case.